scrapy/http/request/json_request.py

```diff
@@ -0,0 +1,31 @@
+"""
+This module implements the JSONRequest class which is a more convenient class
+(than Request) to generate JSON Requests.
+
+See documentation in docs/topics/request-response.rst
+"""
+
+import json
+import warnings
+
+from scrapy.http.request import Request
+
+
+class JSONRequest(Request):
+    def __init__(self, *args, **kwargs):
+        body_passed = kwargs.get('body', None) is not None
+        data = kwargs.pop('data', None)
+        data_passed = data is not None
+
+        if body_passed and data_passed:
+            warnings.warn('Both body and data passed. data will be ignored')
+
+        elif not body_passed and data_passed:
+            kwargs['body'] = json.dumps(data)
```

Comment:
I'm not 100% sure here, asking for feedback, based on discussion with @kmike 
Shall we use ``sort_keys=True`` for ``json.dumps``? The reason to use them is to improve duplicate filter behaviour, without ``sort_keys``, it can happen that two requests for which ``data`` are equal, will get encoded to different bodies, and will not be filtered out.

Reply:
@lopuhin Agree. Changed this.